product_supplierinfo_for_customer/models/product_product.py

```diff
@@ -64,34 +64,22 @@ def _get_price_from_customerinfo(self, partner_id):
         self.ensure_one()
         if not partner_id:
             return 0.0
-        customerinfo = self.env["product.customerinfo"].search(
-            [
-                ("name", "=", partner_id),
-                "|",
-                ("product_id", "=", self.id),
-                "&",
-                ("product_tmpl_id", "=", self.product_tmpl_id.id),
-                ("product_id", "=", False),
-            ],
-            limit=1,
-            order="product_id, sequence",
-        )
+        partner = self.env["res.partner"].browse(partner_id)
+        customerinfo = self._select_customerinfo(partner=partner)
         if customerinfo:
             return customerinfo.price
         return 0.0
 
     def price_compute(self, price_type, uom=False, currency=False, company=False):
         if price_type == "partner":
-            partner_id = self.env.context.get(
-                "partner_id", False
-            ) or self.env.context.get("partner", False)
-            if partner_id and isinstance(partner_id, models.BaseModel):
```

Comment:
This polymorphism seems to be still needed (what I was talking about is to not create new ones).